### PR TITLE
[llvm][PowerPC] Correct handling of spill slots for SPE when EXPENSIVE_CHECKS is enabled

### PR DESCRIPTION
**Pull request by DavidSpickett**
This was modifying a container as it iterated it, which tripped a check in libstdc++'s debug checks.

Instead, just assign to the item via the reference we already have.

This fixes the following expensive checks failures on my machine:
  LLVM :: CodeGen/PowerPC/fp-strict.ll
  LLVM :: CodeGen/PowerPC/pr55463.ll
  LLVM :: CodeGen/PowerPC/register-pressure.ll
  LLVM :: CodeGen/PowerPC/spe.ll

Which are some of the tests noted by #68594.

### CODE REVIEW

**Comment by DavidSpickett** [comment]
@arsenm On https://reviews.llvm.org/D152437 you said:
"Don't use reference to MCPhysReg"

I can address that too if you want. Looks like MCPhysReg is passed around by copy is that right?

**Review by kosarev — APPROVED** [review]
Thanks!

**Comment by arsenm** [comment]
> @arsenm On https://reviews.llvm.org/D152437 you said: "Don't use reference to MCPhysReg"
> 
> I can address that too if you want. Looks like MCPhysReg is passed around by copy is that right?

Yes, MCPhysReg is just an int16_t

File: llvm/lib/Target/PowerPC/PPCFrameLowering.cpp [hunk 1]
```diff
@@ -2334,24 +2334,16 @@ bool PPCFrameLowering::assignCalleeSavedSpillSlots(
...
-      if (Lower) {
-        if (MRI.isPhysRegModified(Higher)) {
-          Idx++;
-          continue;
-        } else {
+      if ( // Check only for SuperRegs.
+          Lower &&
           // Replace Reg if only lower-32 bits modified
-          CSI.erase(CSI.begin() + Idx);
-          CSI.insert(CSI.begin() + Idx, CalleeSavedInfo(Lower));
```
[the remark this answers is not in the document]
Comment (DavidSpickett) [reply llvm/lib/Target/PowerPC/PPCFrameLowering.cpp:2351]: For this instance of `assignCalleeSavedSpillSlots` yes, but other instances do modify the vector so it cannot be changed for everyone (https://github.com/llvm/llvm-project/blob/de55a2843fae6afd4b0589d81496096a4ff73cbd/llvm/include/llvm/CodeGen/TargetFrameLowering.h#L153).

**Pull request merged**